### PR TITLE
Class attribute building

### PR DESCRIPTION
**Pull request by BPScott**
Defining classes and attributes in templates makes more sense as then they're much nearer to where they're used. Having to flit between two files to find the css classes that are used in a given case is annoying.

So thus define classes in the presenters.

### CODE REVIEW

**Comment by programmes-developers** [comment]
Can one of the admins verify this patch?

File: src/Ds2013/Molecule/DateList/DateListItemPresenter.php [hunk 1]
```diff
@@ -37,55 +37,36 @@ public function __construct(DateTimeImmutable $datetime, Service $service, int $
...
+        return $this->offset;
     }
 
     public function getServicePid(): string
     {
         return (string) $this->service->getPid();
     }
 
-    public function shouldShowLink(): bool
+    public function isLink(): bool
```
Comment (votemike) [inline src/Ds2013/Molecule/DateList/DateListItemPresenter.php]: What is `isLink()`?

File: src/Ds2013/Molecule/DateList/DateListItemPresenter.php [hunk 2]
```diff
@@ -37,55 +37,36 @@ public function __construct(DateTimeImmutable $datetime, Service $service, int $
...
     }
+
+    public function isGmt(): bool
+    {
+        return $this->options['user_timezone'] == 'GMT';
+    }
+
+    public function isToday(): bool
+    {
+        return ApplicationTime::getTime()->format('Y-m-d') == $this->datetime->format('Y-m-d');
```
Comment (votemike) [inline src/Ds2013/Molecule/DateList/DateListItemPresenter.php]: This doesn't seem to be comparing timezones. What if each time being compared has a different timezone?

File: src/Ds2013/Molecule/DateList/DateListItemPresenter.php (hunk 1 above)
Comment (BPScott) [inline src/Ds2013/Molecule/DateList/DateListItemPresenter.php]: If the current date list item is a link

File: src/Ds2013/Molecule/DateList/DateListItemPresenter.php (hunk 2 above)
Comment (BPScott) [inline src/Ds2013/Molecule/DateList/DateListItemPresenter.php]: Good point. This is the same logic as what was in here previously. The diff is moving the order of the methods so the `is*` methods sit together. Fixing this comparison is out of scope of this PR imo.

Comment (votemike) [inline src/Ds2013/Molecule/DateList/DateListItemPresenter.php]: We can save it for Chronos to deal with 😄
